pyqtgraph/widgets/MultiAxisPlotWidget.py

```diff
@@ -0,0 +1,378 @@
+                    cvb.state["linkedViews"][cvb.XAxis] = weakref.ref(axis_view)
+                    # this signal is received multiple times when using mouse actions directly on the viewbox
+                    # this causes the non top layer views to scroll more than the frontmost one
+                    chart.mapw_conn["axis_view.sigXRangeChanged"] = axis_view.sigXRangeChanged.connect(cvb.linkedXChanged)
+                    chart.mapw_conn["axis_view.sigResized"] = axis_view.sigResized.connect(cvb.linkedXChanged)
+                    # disable autorange on manual movements
+                    chart.mapw_conn["axis_view.sigXRangeChangedManually"] = axis_view.sigXRangeChangedManually.connect(lambda mask: self.disableAxisAutoRange(axis_name))
+                elif axis.orientation in ["right", "left"]:
+                    # connect axis main view changes to view
+                    cvb.state["linkedViews"][cvb.YAxis] = weakref.ref(axis_view)
+                    # this signal is received multiple times when using mouse actions directly on the viewbox
+                    # this causes the non top layer views to scroll more than the frontmost one
+                    chart.mapw_conn["axis_view.sigYRangeChanged"] = axis_view.sigYRangeChanged.connect(cvb.linkedYChanged)
+                    chart.mapw_conn["axis_view.sigResized"] = axis_view.sigResized.connect(cvb.linkedYChanged)
+                    # disable autorange on manual movements
+                    chart.mapw_conn["axis_view.sigYRangeChangedManually"] = axis_view.sigYRangeChangedManually.connect(lambda mask: self.disableAxisAutoRange(axis_name))
+            chart.mapw_conn["cvb.sigStateChanged"] = cvb.sigStateChanged.emit(cvb)
+        # resize plotitem according to the master one
+        # resizing it's view doesn't work for some reason
+        chart.mapw_conn["self.vb.sigResized"] = self.vb.sigResized.connect(lambda vb: chart.plotItem.setGeometry(vb.sceneBoundingRect()))
+        # fix prepareForPaint by outofculture
+        chart.mapw_conn["scene.sigPrepareForPaint"] = scene.sigPrepareForPaint.connect(cvb.prepareForPaint)
+        if cvb is not tvb:
+            # FROM "https://github.com/pyqtgraph/pyqtgraph/pull/2010" by herodotus77
+            # propagate mouse actions to charts "hidden" behind
+            chart.mapw_conn["tvb.sigMouseDragged"] = tvb.sigMouseDragged.connect(cvb.mouseDragEvent)
+            chart.mapw_conn["tvb.sigMouseWheel"] = tvb.sigMouseWheel.connect(cvb.wheelEvent)
+            chart.mapw_conn["tvb.sigHistoryChanged"] = tvb.sigHistoryChanged.connect(cvb.scaleHistory)
+
+    def disconnect_all(self, chart):
+        """Disconnects all signals related to this widget for the given chart."""
+        for conn_name, conn in chart.mapw_conn.items():
+            if conn is not None:
+                try:
+                    self.disconnect(conn)
+                except (TypeError, RuntimeError):
+                    pass
```

Comment:
> 

i can call QObject.disconnect instead of self.disconnect and might be a little cleaner, the point of not using the .disconnect of the signal is to avoid disconnecting the connections the actual user might have connected. This was also suggested here #1359#discussion_r749660542
As for the except block it is there to catch exceptions thrown when trying to disconnect signals not connected to anything.

Reply:
not a code request change, but fyi we apparently tried to abstract this away a bit:

https://github.com/pyqtgraph/pyqtgraph/blob/master/pyqtgraph/functions.py#L3201-L3217

Reply:
oh ok, how can i use this? does this override the QtCore.Signal.disconnect function?

Reply:
oh wait, I just re-read your code, you're using the QMetaObjects already, I would ignore the comment I made above regarding `pg.disconnect`

In your code, instead of doing `self.disconnect(conn)` I think you want to use `pg.Qt.QtCore.QObject.disconnect(conn)`  as @outofculture suggested 👍🏻

Reply:
done